### PR TITLE
google-ai-generativelanguage 0.6.10 :snowflake: 

### PR DESCRIPTION
**Pull request by Andrii-Osipov**
google-ai-generativelanguage 0.6.10 :snowflake: 

**Destination channel:** Snowflake

- [PKG-6066](https://anaconda.atlassian.net/browse/PKG-6066) 
- [source code](https://github.com/googleapis/google-cloud-python/tree/main/packages/google-ai-generativelanguage)
- [PyPi](https://pypi.org/project/google-ai-generativelanguage/)
- [conda-forge](https://github.com/conda-forge/google-ai-generativelanguage-feedstock)

[PKG-6066]: https://anaconda.atlassian.net/browse/PKG-6066?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -10,35 +10,45 @@ source:
   sha256: bfa03788803061c80c005e28fab5d96f465fcf1c7d28a1eefb1631fb4a2ab67f
 
 build:
-  noarch: python
-  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
   number: 0
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  skip: True  # [py<39]
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
  skip: True  # [py<37]
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -10,35 +10,45 @@ source:
...
+    - python
     - pip
     - setuptools
+    - wheel
   run:
-    - python >=3.7
+    - python
     - google-api-core >=1.34.1,<3.0.0dev,!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.*,!=2.4.*,!=2.5.*,!=2.6.*,!=2.7.*,!=2.8.*,!=2.9.*,!=2.10.*
     - google-auth >=2.14.1,<3.0.0dev,!=2.24.0,!=2.25.0
     - proto-plus >=1.22.3,<2.0.0dev
```
Comment (cbouss) [inline recipe/meta.yaml:27]:
```suggestion
    - proto-plus >=1.22.3,<2.0.0dev
    - proto-plus >=1.25.0,<2.0.0dev # [py>=313]
```
https://github.com/googleapis/google-cloud-python/blob/google-ai-generativelanguage-v0.6.12/packages/google-ai-generativelanguage/setup.py

File: recipe/meta.yaml [hunk 3]
```diff
@@ -10,35 +10,45 @@ source:
...
   host:
-    - python >=3.7
+    - python
     - pip
     - setuptools
+    - wheel
   run:
-    - python >=3.7
+    - python
     - google-api-core >=1.34.1,<3.0.0dev,!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.*,!=2.4.*,!=2.5.*,!=2.6.*,!=2.7.*,!=2.8.*,!=2.9.*,!=2.10.*
```
Comment (cbouss) [inline recipe/meta.yaml:25]:
```suggestion
    - google-api-core >=1.34.1,<3.0.0dev,!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.*,!=2.4.*,!=2.5.*,!=2.6.*,!=2.7.*,!=2.8.*,!=2.9.*,!=2.10.*
    - grpcio >=1.33.2,< 2.0dev
    - grpcio >= 1.49.1,< 2.0dev # [py>=311]
    - grpcio-status >=1.33.2,< 2.0dev
    - grpcio-status >= 1.49.1,< 2.0dev # [py>=311]
```
upstreams requires google-api-core[grpc] https://github.com/googleapis/google-cloud-python/blob/google-ai-generativelanguage-v0.6.12/packages/google-ai-generativelanguage/setup.py#L44C6-L44C21

There is no good way to express this. We can however refer to https://github.com/googleapis/google-cloud-python/blob/google-ai-generativelanguage-v0.6.12/packages/google-ai-generativelanguage/setup.py#L44C6-L44C21 and https://inspector.pypi.io/project/google-api-core/2.23.0/packages/fa/6b/b98553c2061c4e2186f5bbfb1aa1a6ef13fc0775c096d18595d3c99ba023/google_api_core-2.23.0.tar.gz/google_api_core-2.23.0/pyproject.toml#line.62

File: recipe/meta.yaml [hunk 4]
```diff
@@ -10,35 +10,45 @@ source:
...
+    - tests
   imports:
     - google
   commands:
     - pip check
+    - pytest -v tests
   requires:
     - pip
+    - pytest
+    - grpcio
```
Comment (cbouss) [inline recipe/meta.yaml:41]:
```suggestion
```
This should be a run dependency.

File: recipe/meta.yaml [hunk 5]
```diff
@@ -10,35 +10,49 @@ source:
...
     - setuptools
+    - wheel
   run:
-    - python >=3.7
+    - python
     - google-api-core >=1.34.1,<3.0.0dev,!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.*,!=2.4.*,!=2.5.*,!=2.6.*,!=2.7.*,!=2.8.*,!=2.9.*,!=2.10.*
+    - grpcio >=1.33.2,<2.0dev
+    - grpcio >=1.49.1,<2.0dev # [py>=311]
+    - grpcio-status >=1.33.2,<2.0dev
+    - grpcio-status >=1.49.1,<2.0dev # [py>=311]
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
    - google-api-core-grpc >=1.34.1,<3.0.0dev,!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.*,!=2.4.*,!=2.5.*,!=2.6.*,!=2.7.*,!=2.8.*,!=2.9.*,!=2.10.*
```
Sorry, I didn't check yesterday, but we actually provide a package google-api-core-grpc which has he correct dependencies set.

**Review by skupr-anaconda — APPROVED** [review]
LGTM but Charle's comment

**Review by boldorider4 — APPROVED** [review]

File: recipe/meta.yaml (hunk 5 above)
Comment (boldorider4) [reply recipe/meta.yaml]: these dependencies are not applicable to version `0.6.10`, which is what we need for https://github.com/AnacondaRecipes/google-generativeai-feedstock/pull/1

**Pull request merged**